specification/synapse/resource-manager/Microsoft.Synapse/preview/2021-06-01-preview/workspace.json

```diff
@@ -931,6 +931,14 @@
         "filesystem": {
           "description": "Filesystem name",
           "type": "string"
+        },
+        "resourceId": {
+          "description": "ARM resource Id of this storage account",
+          "type": "string"
+        },
+        "createManagedPrivateEndpoint": {
```

Comment:
>createManagedPrivateEndpoint [](http://example.com/codeflow?start=9&length=28)

Consider using a string enum, e.g. managedPrivateEndpointHandling: { Create, None } or similar. Generally enums are more descriptive, flexible and future-proof than booleans and easier for customers to discover, understand and use.

Reply:
This feature is only meant to be used at the time of creating a Synapse workspace. When the user selects the checkbox, this feature helps create a private endpoint to a special storage account automatically. Once created, the user cannot modify this private endpoint so there is no further management possible, other than deleting it. But we automatically delete the private endpoint even today, when the user deletes the Synapse workspace. So I would say that all in all, there is no need to change this bool to an enum in this particular case.